backend/prompt_studio/prompt_studio_index_manager/migrations/0002_remove_incorrect_index_keys.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 4.2.1 on 2024-04-24 14:03
+
+import logging
+from typing import Any
+
+from django.db import migrations, models
+
+logger = logging.getLogger(__name__)
+
+# This migration flushes all records in IndexManager which have an incorrect index
+# key generated from an older implementation
+# Refer https://github.com/Zipstack/unstract-sdk/blob/1c61f4325a38e1bd25e207ca2c426971aa7b62cf/src/unstract/sdk/index.py#L322C1-L349C10  # noqa E501
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("prompt_studio_index_manager", "0001_initial"),
+    ]
+
+    def remove_incorrect_index_keys(apps: Any, schema_editor: Any) -> None:
+        IndexManager = apps.get_model("prompt_studio_index_manager", "IndexManager")
+
+        incorrect_index_pattern = (
+            r"[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}\|"  # Tool ID
+            r"[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}\|"  # Vector DB
+            r"[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}\|"  # Embedding
+            r"[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}\|"  # X2Text
+            r"\d+\|\d+\|"  # Chunk size and chunk overlap
+            r"[\da-f]{64}"  # SHA-256 file hash
+        )
+
+        incorrect_indexes = IndexManager.objects.filter(
+            models.Q(raw_index_id__regex=incorrect_index_pattern)
+            | models.Q(summarize_index_id__regex=incorrect_index_pattern)
+        )
+        logger.info(f"Deleting index manager records: {incorrect_indexes}")
+        incorrect_indexes.delete()
+        return
+
+    operations = [
+        migrations.RunPython(
+            remove_incorrect_index_keys, reverse_code=migrations.RunPython.noop
+        ),
+    ]
```

Comment:
@chandrasekharan-zipstack quick question. I believe we are removing the keys only from DB. What about the files indexed in the vector db? It will just stay there as a stale entry right?

PS: This might not cause any issues and it can be okay to leave it as a stale entry. Want to make sure this point is thought about and discussed if needed. 
cc: @hari-kuriakose

Reply:
@ritwik-g yes the entries in the vector DB would remain as stale entry. With the current code as well, deleting documents does not ensure that index entries in vector DB is deleted. This is planned as a separate task altogether. @nehabagdia is also aware of this stale entry presence

Reply:
@chandrasekharan-zipstack Ideally we need to remove stale entries in Vector DB too.

Reply:
Answered [here](https://github.com/Zipstack/unstract/pull/289#issuecomment-2084481029)